spk/firebird/src/service-setup.sh

```diff
@@ -0,0 +1,105 @@
+        exit 1;    
+    };
+}
+
+gen_password() {
+    head -c20 /dev/urandom | base64
+}
+
+service_postinst () {
+    cd "$SYNOPKG_PKGDEST"
+    SET_PASSWD=false
+    SECDB_CREATED=false
+
+    # =================== Build Messages ========================
+    # Build the firebird.msg file during both install and upgrade.
+    # The messages differ across versions and must always match the binaries
+    # version.
+    # Building the messages should be first step because the tools used here
+    # later are already expectine the message file to exist.
+    bin/gbak -rep msg.gbak msg.fdb
+    bin/build_file
+
+    # Publish the message file so that other users can use isql
+    # (isql and other cli tools access the file to translate sql errors)
+    chmod go+r firebird.msg
+
+    echo "Remove unnecessary intermediate files"
+    rm -f bin/build_file msg.gbak msg.fdb
+
+    # =================== Build Security Database ===============
+    # Build security database only when none is present yet or we are being
+    # installed
+    if [ ! -f "$FB_SECDB" ] || [ "$SYNOPKG_PKG_STATUS" = "INSTALL" ];then
+        bin/gbak -rep security4.gbak "$FB_SECDB"
+
+        # We just built the security database. We must also set the password.
+        SECDB_CREATED=true
+    fi
+    
+    # We already built the DB. This is no longer needed.
+    rm -f security4.gbak
+
+    # =================== Set SYSDBA password ===================
+    #
+    # Setting the SYSDBA password is somewhat tricky:
+    # - When being installed from the UI, the password is provided by the wizard
+    # - When being installed from cli, no wizard runs and we get empty string
+    #       in such case we generate new password and announce it in the log.
+    # - When upgrading, we usually don't have to do anything, unless the user
+    #       deleted the security database and is re-installing the package
+    #       to fix it. Then:
+    #       o We either use the password from SYSDBA.password file (if it exists)
+    #       o Or generate new random password and announce it in the log.
+    # - Upgrades where security.db exists are best because we don't do anything.
+    #
+    if [ "$SYNOPKG_PKG_STATUS" = "INSTALL" ]; then
+        if [ "x$wizard_sysdba_password" = "x" ]; then
+            # User provided new password. We override the password file even
+            # if it exists because the user expects to have the new password.
+            gen_password > "$SYSDBA_PASSWORD_FILE";
+        else
+            # Use the users password
+            printf "%s" "$wizard_sysdba_password" > "$SYSDBA_PASSWORD_FILE"
+        fi
+        SET_PASSWD=true
+    else #UPGRADING
+        if $SECDB_CREATED && [ ! -f "$SYSDBA_PASSWORD_FILE" ]; then
+            # We are upgrading/re-installing and the user deleted 
+            # both the password file and the database file
+            gen_password > "$SYSDBA_PASSWORD_FILE";
+            SET_PASSWD=true
+        fi
```

Comment:
IMHO `SET_PASSWORD=true` must always be set when SECDB_CREATED is true.
This must not depend on non existing password file.

Otherwise the usecase "delete security db to recreate it without deletion of the password file" would not work (i.e. the password is not set).
Since upgrade is not allowed with deleting both security db and password file this if clause will never be executed and can be removed (when SET_PASSWD=true is set when creating the security db).